### DISCUSSION TITLE
Issue with python-snap7 installation on Kali Linux

### DISCUSSION THREAD

**Discussion by iambeluchi** [gijzelaerr/python-snap7#265] [Q&A]
No Valid OpenPGP data found.

I got this error message while installing on Kali Linux.

Please I need help resolving this. I have tried all I could
![Screenshot 2021-03-25 at 09 40 06](https://user-images.githubusercontent.com/22841347/112514635-ce8e2e00-8d95-11eb-9ee3-ea4fc0269e82.png)

**Comment by 0xmra** [comment]
are you trying to install packages for the snap7 library.

To install you can try:
```bash
 $ sudo add-apt-repository ppa:gijzelaar/snap7
 $ sudo apt-get update
 $ sudo apt-get install libsnap7-dev libsnap7-1
```

**Reply by swamper123** [reply]
> To install you can try:
 $ sudo add-apt-repository ppa:gijzelaar/snap7

Isn't that exactly what he tried in the first line?

**Reply by gijzelaerr** [reply]
i dont see any output for the last 2 commands in the copy-paste.

**Reply by swamper123** [reply]
I think @iambeluchi is worrying about the: _No Valid OpenPGP data found._ and hasn't proceed further (or even can't?).
Maybe a proxy thing? Or is it, because Kali is debian based and (I guess) ppa is a typical Ubuntu thing?

**Comment by gijzelaerr** [comment]
my guess @ambeluchi is confused by the warning. It looks like it successfully imported the key, and it should be possible to install the snap7 package now.

**Reply by iambeluchi** [reply]
I wasn't able to install it still... I switched to Ubuntu now... Its all working fine

**Comment by swamper123** [comment]
@iambeluchi A workaround can be, that you install snap7 directly from [sourceforge ](https://sourceforge.net/projects/snap7/files/) (choose the 1.4.2. folders, the "Download" button has ...some other stuff as default), choose the right release version for your device and install it. Don't forget executing ldconfig() in terminal!
Then install the python-snap7 as normal and ecerything should be fine.

**Comment by iambeluchi** [comment]
I have installed on Ubuntu... However, I need to know if this can also work on Mac.

**Reply by gijzelaerr** [reply]
you can install snap7 using homebrew on OSX.

**Comment by iambeluchi** [comment]
I tried with a friend last week and it was a bit complicated... We were able to install the wrapper.., but not the main snap7 before wrapper....
